### DISCUSSION TITLE
Bookmarks gone (some, not all)

### DISCUSSION THREAD

**Discussion by GutHib** [organicmaps/organicmaps#4900] [Q&A]
I'm having the problems that all the bookmarks I set yesterday are gone... In the configuration, all lists are checked, though. Here's what probably happened:

- Bookmark category changed to a list that contains a track instead of individual points. Why this happened (did it happen on its own or did I do it accidentially) I have no clue.
- After setting the bookmarks, they were clearly visible on the map.
- Today they are gone, and the file they were supposedly saved in doesn't have an updated save date, either.

Even though my hopes are limited, I would like to recover the bookmarks - if I fail, I might at least find out what happened so this won't happen again.

If the bookmarks weren't actually saved in that particular kml file, while did they show on the map in the first place? Is there some sort of temporary file and are bookmarks only saved for good upon exit?

Thanks!

**Comment by biodranik** [comment]
Please enable logs in OM settings, restart the app, and send us logs via the Help dialog => Report a bug. If the file was corrupted due to some non-valid XML symbol (see, for example, #3837), we'll restore it from logs for you.

**Comment by GutHib** [comment]
Ok, but if the save date didn't change, then that doesn't point to file corruption, or does it?

Also, if I enable logs now, will it still help to figure out what happened yesterday?

**Reply by biodranik** [reply]
As I don't know what you were doing _exactly_, I can only guess. There is an issue when the KML file may not load if you previously saved some bad symbols in any bookmark's description.

I do not know how the track appeared in your bookmarks. Did you import some files with a track?

In any case, you can either check the app's folder yourself (if you have root access), or enable logs and send them to us, maybe logs will show some corruption on a KML file, and some or all bookmarks can be restored. Or you can read logs yourself )

**Comment by GutHib** [comment]
Sorry if I was unclear... I do have root access and checked the files in /data/user/0/app.organicmaps/files/bookmarks already - that's how I know there's no fresh "changed" date.

And I'm fully cooperative. :) But to me, if the file didn't save, it seems like a waste of your time to analyze it. Am I wrong?

I created the track file externally (via some website). If Organic Maps sees the file as corrupted in a way that it will display its contents, but doesn't want to save to it, then fine. But to me, it doesn't explain why all the bookmarks I created yesterday remained visible without obvious problems, only to vanish the next morning. If the bookmark can't be saved, shouldn't it go up in flames immediately - or does it live in some cache until OM is restarted?

EDIT: Ok, with the help of the logs I have it figured out. After a reinstall, I got the permissions of the bookmarks folder wrong. And yes, OM keeps the bookmarks in memory, so if there's a write error it will only show once OM is restarted. Thanks for the assistance.

**Reply by biodranik** [reply]
How was that possible to get the wrong permission for the bookmarks folder? Did you break it manually? Should we find and fix a bug that caused it?

OM's code relies on the Android system that _guarantees_ that the folder/file in the internal memory will be writable.

**Reply by GutHib** [reply]
No worries, the permission issues were my own personal screwup.

An error message would have been nice, though (given the fact that OM can literally be a life-saving app).
